content/en/docs/concepts/policy/pid-limiting.md

```diff
@@ -0,0 +1,43 @@
+---
+reviewers:
+- derekwaynecarr
+title: PID Limiting
+content_type: concept
+weight: 10
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.20" state="stable" >}}
+
+PID Limiting features allow administrators are able to provide pod-to-pod PID isolation by defaulting the number of PIDs per pod. In addition, administrators can enable node-to-pod PID isolation by reserving a number of allocatable PIDs to user pods via node allocatable.
+
+<!-- body -->
+
+Process IDs (PIDs) are a fundamental resource on Linux hosts. It is trivial to hit the task limit without hitting any other resource limits and cause instability to a host machine. Administrators require mechanisms to ensure that user pods cannot induce PID exhaustion that prevents host daemons (runtime, kubelet, etc) from running. In addition, it is important to ensure that PIDs are limited among pods in order to ensure they have limited impact to other workloads on the node.
+
+You can configure a kubelet to limit the number of PIDs a given pod can consume. If that machine supports `32,768` PIDs and `100` pods, one can give each pod a budget of `300` PIDs to prevent total exhaustion of PIDs. If the admin wants to overcommit PIDs similar to cpu or memory, they may do so as well with some additional risks. Either way, no one pod can bring the whole machine down. This will generally prevent against simple fork bombs from taking over your cluster.
```

Comment:
```suggestion
You can configure a kubelet to limit the number of PIDs a given pod can consume. For example, if your node's host OS is set to use a maximum of 262144 PIDs and 250 pods, one can give each pod a budget of 1000 PIDs to prevent using up that node's overall number of available PIDs. If the admin wants to overcommit PIDs similar to CPU or memory, they may do so as well with some additional risks. Either way, no one pod can bring the whole machine down. This kind of resource limiting helps to protect against simple fork bombs from affecting service cluster-wide.
```